### DISCUSSION TITLE
Unable to set up codebase on my machine

### DISCUSSION THREAD

**Discussion by VishalGoel11** [checkstyle/checkstyle#16019] [Q&A]
Hi Everyone, Hope everything is going well for all of you.
I want to setup this codebase in my local machine. But unable to do so. Kindly hel me in this regard.

Steps  I follow : 
1.) Fork The repo .
2.) Then clone it to local machine.
3.) Then open it in intelliz idea.
4.) Wait for a time till all dependencies load
5.)Then run it.

**Error : java: package com.puppycrawl.tools.checkstyle.grammar.javadoc does not exist**

**Compile time error -> import com.puppycrawl.tools.checkstyle.grammar.javadoc.JavadocParser at this line.**

I verified the corresponding package didn't contain the mentioned  java file in mentioned package.

 **How will I get this package in this project ??**

Please let me know that anything else need from my side.

**Comment by romani** [comment]
https://checkstyle.org/beginning_development.html

https://github.com/checkstyle/checkstyle/blob/master/.github/CONTRIBUTING.md

**Reply by romani** [reply]
@VishalGoel11, forgot to mention that on first link above, there is link YouTube album on how to setup IDEA from scratch and create first PR, please watch it.

**Reply by romani** [reply]
https://youtube.com/playlist?list=PLHM9s_lN4X0hzOQ0sUmGdroxW0HfREAqj&si=Jbkh-Dpa3QrDkAHO

**Comment by Zopsss** [comment]
> 3.) Then open it in intelliz idea.
4.) Wait for a time till all dependencies load

From: https://checkstyle.org/beginning_development.html

> 4. Before opening project in IDE, build the project in your terminal to download
all needed artifacts. From the repository root folder, run:
>
>`mvn install`

You need to run `mvn install` command before opening the project in IDE.

**Reply by rnveach** [reply]
> You need to run mvn install command before opening the project in IDE.

This doesn't need to be done for Eclipse. I would find it odd if IDEs couldn't handle this properly now.

`mvn test-compile` might be enough if all this is to just get dependencies for both main and test.
